packages/service-library/src/servicelib/rabbitmq.py

```diff
@@ -93,19 +48,45 @@ class RabbitMQClient:
     _rpc_channel: aio_pika.abc.AbstractChannel | None = None
     _rpc: RPC | None = None
 
-    def __post_init__(self):
+    _bad_state: bool = False
+
+    def __post_init__(self) -> None:
         # recommendations are 1 connection per process
         self._connection_pool = aio_pika.pool.Pool(
-            _get_connection, self.settings.dsn, self.client_name, max_size=1
+            self._get_connection, self.settings.dsn, self.client_name, max_size=1
         )
         # channels are not thread safe, what about python?
         self._channel_pool = aio_pika.pool.Pool(self._get_channel, max_size=10)
 
+    async def _get_connection(
+        self, rabbit_broker: str, connection_name: str
+    ) -> aio_pika.abc.AbstractRobustConnection:
+        # NOTE: to show the connection name in the rabbitMQ UI see there
+        # https://www.bountysource.com/issues/89342433-setting-custom-connection-name-via-client_properties-doesn-t-work-when-connecting-using-an-amqp-url
+        #
+        url = f"{rabbit_broker}?name={get_rabbitmq_client_unique_name(connection_name)}"
+        connection = await aio_pika.connect_robust(
+            url,
+            client_properties={"connection_name": connection_name},
+        )
+        connection.close_callbacks.add(
+            functools.partial(connection_close_callback, client=self)
+        )
+        return connection
+
+    @property
+    def bad_state(self) -> bool:
+        return self._bad_state
```

Comment:
where are you changing this state???  aha, so `rabbit_utils.connection_close_callback`, ...  are kind of "friends"
this is a bit unusual. Perhaps document in this client class so it is not misleading

Sugguestion: What about calling it  `_healthy` instead of `bad_state`? This way at least we can connect with the healthcheck concept